### ISSUE TITLE
Unminified output

### ISSUE THREAD

**Issue by ghost** [weavejester/hiccup#65]
Is there a way making hiccup generate unminified output so it would be easier to debug in the browser?

**Comment by dwhite44** [comment]
It would definitely be nice to have some minify options with hiccup. I don't necessarily need unminified but I would like an option to have a space between each element to help with spacing problems that can result from having no whitespace between elements such as this bootstrap issue: https://github.com/twbs/bootstrap/issues/14401

**Comment by mgerlach-klick** [comment]
I wrote this library to deal with this: https://github.com/px0/clj-beautify-web
I usually put something like this in the render chain after the `html` macro:
```clj
(fn [html]
    (if (= "production" (:environment render-params))
    html
    (beautify-web.core/beautify-html html)))
```

**Comment by didibus** [comment]
Is this planned for 2.0 ?

**Comment by weavejester** [comment]
No, plans for 2.0 were scaled back. Adding it in for 2.1 is a possibility, but I haven't had much time to work on Hiccup recently.

**Comment by didibus** [comment]
Ah I see. Okay, thank you, I'll look into beautify-html for my debugging purpose.

**Comment by metametadata** [comment]
I'm also interested in such a built-in feature as I find it more convenient for unit tests and sometimes look at the source code instead of a browser inspector. The current workaround is to use Jericho to postformat the generated HTML:

```clj
[net.htmlparser.jericho/jericho-html "3.4"]
...
(:import [net.htmlparser.jericho Source])
...
(defn formatted-html
  [s]
  (str (-> (.getSourceFormatter (Source. s))
           (.setIndentString "  "))))
```
